### PR TITLE
Add org.jupyter.JupyterLab

### PR DESCRIPTION
**Pull request by kpthedev**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - Since this is an IDE, `filesystem=home` is needed for editing files, opening project directories, and using Python environments.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
  - https://github.com/jupyterlab/jupyterlab-desktop/issues/580
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by kpthedev** [comment]
bot, build org.jupyter.JupyterLab

File: org.jupyter.JupyterLab.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - rm -f control.tar.gz data.tar.xz debian-binary
+      - cp -r opt/* usr/* $FLATPAK_DEST
+      - rm -rf opt usr
+      - install -Dm 755 run.sh $FLATPAK_DEST/bin/
+      - desktop-file-edit --set-key=Exec --set-value="run.sh %U" $FLATPAK_DEST/share/applications/jupyterlab-desktop.desktop
+    sources:
+      - type: file
+        only-arches:
+          - x86_64
+        url: https://github.com/jupyterlab/jupyterlab-desktop/releases/download/v3.6.1-2/JupyterLab-Setup-Debian.deb
```
Comment (hfiguiere) [inline org.jupyter.JupyterLab.yaml:43]: You still build a aarch64 flatpak.

File: org.jupyter.JupyterLab.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+runtime-version: "22.08"
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: "22.08"
+command: run.sh
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline org.jupyter.JupyterLab.yaml:12]: does it support wayland?

File: org.jupyter.JupyterLab.yaml (hunk 1 above)
Comment (kpthedev) [reply org.jupyter.JupyterLab.yaml:43]: Yes, I saw. I've added a `flathub.json` .

File: org.jupyter.JupyterLab.yaml (hunk 2 above)
Comment (kpthedev) [reply org.jupyter.JupyterLab.yaml]: It doesn't; I've removed the permission.

**Comment by kpthedev** [comment]
bot, build org.jupyter.JupyterLab

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: any chance to have it on aarch64? Like building from source?

Comment (JakobDev) [reply flathub.json:2]: It is Electron, so probably not

**Pull request closed without merging**